### ISSUE TITLE
Seemingly redundant alias is required when selecting from the right side of a join

### ISSUE THREAD

**Issue by cpcloud** [ClickHouse/ClickHouse#66133]
**Company or project name**

The Ibis project: https://ibis-project.org

**Describe the unexpected behaviour**

This query fails due to not being able to find `o.a`, but it's not clear why:

```sql
WITH t AS (SELECT 1 AS "a")
SELECT
  "o"."a"
FROM (
  SELECT
    "w"."a"
  FROM "t" AS "s"
  INNER JOIN "t" AS "w"
  USING ("a")
) AS "o"
```

whereas simply adding a redundant alias for `"w"."a"` causes the query to work:

```sql
WITH t AS (SELECT 1 AS "a")
SELECT
  "o"."a"
FROM (
  SELECT
    "w"."a" AS "a"
  FROM "t" AS "s"
  INNER JOIN "t" AS "w"
  USING ("a")
) AS "o"
```

Selecting `"s"."a"` instead of `"w"."a"` (with or without a redundant alias) also works.

**How to reproduce**
* Which ClickHouse server version to use: 24.6.1.4423
* Which interface to use, if matters: `clickhouse local` CLI

**Expected behavior**

I would expect the query to execute and return `1`.

**Error message and/or stacktrace**

```
Code: 47. DB::Exception: Identifier 'o.a' cannot be resolved from subquery with name o. In scope WITH t AS (SELECT 1 AS a) SELECT o.a FROM (SELECT w.a FROM t AS s INNER JOIN t AS w USING (a)) AS o. Maybe you meant: ['w.a']. (UNKNOWN_IDENTIFIER)
```

**Comment by mscdex** [comment]
FWIW I also ran into this issue, except it was with multiple columns and only some of the SELECTed columns had aliases. The solution was to give all columns aliases, even if the name was exactly the same. That allowed the query to be successfully executed.

**Comment by clickgapai** [comment]
_Found via ClickGap automated review. Please close or comment if this is incorrect or needs adjustment._

### TL;DR
**Seemingly redundant alias is required when selecting from the right side of a join**

**ClickGap verified:** reproduced it with the script below on `master`; tested 6 version(s) — 6 affected, 0 clean (see Affects); long-standing (not a recent regression); introducing PR not yet pinned.

### Reproduction
```
-- Expected: returns 1. Actual: fails with UNKNOWN_IDENTIFIER 'o.a'.
WITH t AS (SELECT 1 AS "a")
SELECT
  "o"."a"
FROM (
  SELECT
    "w"."a"
  FROM "t" AS "s"
  INNER JOIN "t" AS "w"
  USING ("a")
) AS "o";

-- Workarounds that DO return 1 (show the asymmetry):
--   inner SELECT "w"."a" AS "a"   (explicit redundant alias)
--   inner SELECT "s"."a"           (left-side qualifier)
-- The unaliased right-side reference SELECT "w"."a" names the
-- subquery column "w.a" instead of the merged USING key "a".
```
[▶ Run on ClickHouse Fiddle](https://fiddle.clickhouse.com/9f998276-71de-4fed-8aac-a7e4b8985fc1)

### Bisect
- Reproduces on every release line tested down to `v24.1.8.22` (the oldest with a downloadable binary for this repro). The introducing change predates `v24.1.8.22`, so it can't be pinned from release binaries — this is a **long-standing** issue, not a recent regression. A maintainer with full git history can `git blame` the relevant file/line to find the exact change.

### Affects
- **Reproduces on**: `master, 26.5, 26.4, 26.3, 26.2, 26.1`

### Suggested next step
Manual triage of the `comp-query-analyzer` path is needed. The reproducer above should be a fast first check.

**Component:** `comp-query-analyzer`
**Severity:** `P2`

---
_ClickGap · Finding: `h_issue_66133`_




### Triage metadata
- Primary component: `comp-query-analyzer`
- Secondary components: `comp-joins`
- Component owners (`comp-query-analyzer`): @KochetovNicolai @novikd
- Possible duplicate: #97902 — Both report the same query-analyzer/JOIN bug where selecting a joined column from a subquery fails identifier resolution unless a redundant alias is added.
- Related: #94558
